include/crm/common/xml_internal.h

```diff
@@ -235,4 +235,7 @@ pcmk__xe_next(const xmlNode *child)
     return next;
 }
 
+void
+pcmk__xml_set_propv(xmlNodePtr node, va_list pairs);
```

Comment:
I recently started a naming convention for XML functions depending on what kind of xmlNode they take. If it can take any kind of node (comments, cdata, etc.), then it should start with pcmk__xml. If it can only operate on XML element nodes, it should start with pcmk_xe.

The idea is that we sometimes have issues where there's an XML comment somewhere and some code chokes on it because it assumes all child nodes are elements. The hope is that if it's more obvious what the function operates on, we'll make fewer of those mistakes.

However in this case we have a further issue. The new functions as written can only be used with output XML, not with CIB XML, because they do not enforce ACLs. ACLs are the main reason we have crm_xml_add().

I'm pretty sure at this point we don't have much distinction between CIB XML and non-CIB XML. Aside from the output code, we tend to use the ACL-aware XML functions with everything, even when ACLs don't matter (e.g. IPC messages).

I'm not sure what's the best way forward. We could have separate functions for ACL-aware XML and non-ACL-aware XML, or we could use the ACL-aware functions all the time, or maybe we could have some document flag for whether the ACL logic is needed.